### PR TITLE
Add access_denied error paragraph

### LINKED ISSUE #89: Need better error description and examples for more common cases
The current specification does not provide error codes and examples of cases where the wallet does not have the credentials to answer a request, or when the user does not give consent to share the credentials with the RP.

We need to provide examples. 

While OID4VP allows to provide an error in that case, we should probably discuss whether it makes sense to introduce additional error codes for these more common cases:
- user did not provide consent or rejected the request.
- user does not have all the credentials in their wallet to answer the request.

Thoughts?

### PR DESCRIPTION
**Pull request by awoie**
This PR does:
- Fixes #89 
- Adds `access_denied` error to Response Error section

This PR is non-breaking.

### CODE REVIEW

File: openid-4-verifiable-presentations-1_0.md [hunk 1]
```diff
@@ -671,6 +671,12 @@ The error response follows the rules as defined in [@!RFC6749], with the followi
 
 Usage of `client_metadata` or `client_metadata_uri` parameters with `client_id` that the Wallet might be seeing for the first time is mutually exclusive with the registration mechanism where Self-Issued OP assigns `client_id` to the Verifier after receiving Verifier metadata.
 
+`access_denied`:
+
+- The Wallet did not have the requested Credentials to satisfy the Authorization Request.
+- The end-user did not give consent to share the requested Credentials with the Verifier.
+- The Wallet failed to authenticate the end-user.
```
Comment (Sakurann) [inline openid-4-verifiable-presentations-1_0.md:678]:
```suggestion
- The End-User did not give consent to share the requested Credentials with the Verifier.
- The Wallet failed to authenticate the End-User.
```

File: openid-4-verifiable-presentations-1_0.md [hunk 2]
```diff
@@ -671,6 +671,12 @@ The error response follows the rules as defined in [@!RFC6749], with the followi
 
 Usage of `client_metadata` or `client_metadata_uri` parameters with `client_id` that the Wallet might be seeing for the first time is mutually exclusive with the registration mechanism where Self-Issued OP assigns `client_id` to the Verifier after receiving Verifier metadata.
 
+`access_denied`:
+
+- The Wallet did not have the requested Credentials to satisfy the Authorization Request.
```
Comment (Sakurann) [inline openid-4-verifiable-presentations-1_0.md:676]: is this really `access_denied`..?

Comment (awoie) [reply openid-4-verifiable-presentations-1_0.md:676]: What other error would you suggest? I would not create a dedicated error for privacy reasons.

**Review by jogu — APPROVED** [review]

File: openid-4-verifiable-presentations-1_0.md (hunk 2 above)
Comment (Sakurann) [reply openid-4-verifiable-presentations-1_0.md:676]: yeah, i don't really know an alternative, `access_denied` feels good enough after the WG discussion.

**Review by Sakurann — APPROVED** [review]

**Review by c2bo — APPROVED** [review]

**Review by David-Chadwick — APPROVED** [review]

**Pull request merged**